### PR TITLE
Added Python 3 compatibility

### LINKED ISSUE #70: Python 3 support
If I did the work to submit a PR to provide for Python 2 and 3 support, would it be accepted?

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #70

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @noisecapella! I've created [OSPR-1216](https://openedx.atlassian.net/browse/OSPR-1216) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/opaque-keys/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by noisecapella** [comment]
I'm not sure if this is done yet but I got the tests passing. Any comments so far?

**Comment by clintonb** [comment]
Ironically, we test all of the bad cases, but don't have a test as simple as ensuring `CourseKey.from_string('course-v1:MITx+0.111x+2T2015')` returns successfully. Sadly, it does not. I'm investigating to figure out what's going on with `stevedore`.

@cpennington any suggestions?

**Comment by clintonb** [comment]
Here is a stack trace:
```
Traceback (most recent call last):
  File "/Users/cblackburn/.virtualenvs/course-discovery/lib/python3.5/site-packages/opaque_keys/__init__.py", line 229, in get_namespace_plugin
    return drivers[namespace].plugin
  File "/Users/cblackburn/.virtualenvs/course-discovery/lib/python3.5/site-packages/stevedore/extension.py", line 284, in __getitem__
    return self._extensions_by_name[name]
KeyError: 'course-v1'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py", line 2407, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/Applications/PyCharm.app/Contents/helpers/pydev/pydevd.py", line 1798, in run
    launch(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc) 
  File "/Users/cblackburn/Library/Preferences/PyCharm50/scratches/scratch_46", line 3, in <module>
    CourseKey.from_string('course-v1:MITx+0.111x+2T2015')
  File "/Users/cblackburn/.virtualenvs/course-discovery/lib/python3.5/site-packages/opaque_keys/__init__.py", line 186, in from_string
    return cls.get_namespace_plugin(namespace)._from_string(rest)
  File "/Users/cblackburn/.virtualenvs/course-discovery/lib/python3.5/site-packages/opaque_keys/__init__.py", line 234, in get_namespace_plugin
    raise InvalidKeyError(cls, u'{}:*'.format(namespace))
opaque_keys.InvalidKeyError: <class 'opaque_keys.edx.keys.CourseKey'>: course-v1:*

```

**Comment by clintonb** [comment]
Backtracking...we are going to go a different route for our purposes, and not use opaque-keys. Nevertheless, I am happy to review once this issue is resolved (by someone else).

**Comment by noisecapella** [comment]
I'll add a test when I get back to work on this later. When you say you're going a different route, do you mean that edX doesn't have an interest in this library in general or is that specific to your project?

**Comment by clintonb** [comment]
I meant that, for my specific use case I was going to use an alternative solution. However, that solution may not pan out, so we will probably be fixing this.

**Comment by clintonb** [comment]
Correction: I checked out the wrong branch. I have confirmed this code works as expected for parsing course key strings. @noisecapella would you squash commits, please?

**Comment by clintonb** [comment]
:+1: once commits squashed.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[tox]
+envlist = py27,py34
```
Comment (clintonb) [inline tox.ini]: Use py35 rather than py34, unless you have a specific need for Python 3.4.

File: test-requirements.txt [hunk 2]
```diff
@@ -5,3 +5,8 @@ mock
 nose
 pep8
 sphinx
+pytest
```
Comment (clintonb) [inline test-requirements.txt]: Is there a reason you are switching everything to pytest instead of sticking with unittest?

File: .coveragerc [hunk 3]
```diff
@@ -1,3 +1,4 @@
 [run]
 branch = True
-source = ./opaque_keys
\ No newline at end of file
+source = ./opaque_keys
+omit = ./.tox/*
```
Comment (clintonb) [inline .coveragerc]: Is this necessary? `./.tox` is not a subdirectory of `./opaque_keys`.

**Comment by noisecapella** [comment]
Thanks for the quick review. I'm on vacation until Monday but I'll try to address your comments soon after

**Comment by clintonb** [comment]
@noisecapella we will need this sooner rather than later. Do you mind if I take over getting this merged?

File: .coveragerc (hunk 3 above)
Comment (clintonb) [inline .coveragerc]: Never mind. This is needed.

**Comment by clintonb** [comment]
@noisecapella thanks for your contributions! These changes have been released as [version 0.3.0](https://github.com/edx/opaque-keys/releases/tag/0.3.0), which should be on [PyPI](https://pypi.python.org/pypi/edx-opaque-keys) shortly.

**Comment by noisecapella** [comment]
Thanks for the merge!

**Pull request merged**